src/server/GameController.cpp

```diff
@@ -22,33 +22,58 @@ GameController::~GameController() {
     //de-construcstructor
 }
 
-void GameController::update(int client_id, sf::Vector2f client_direction) {
-    // Update players
-    mPlayers[client_id]->setDirection(client_direction);
+void GameController::update() {
+    // Get clients
+    std::vector<int> client_ids = mNetworkingServer->getClientIds();
+
+    // Create new players
+    for (const int client_id : client_ids) {
+        if (!mPlayers[client_id]->isActive()) {
+            createPlayer(client_id);
+        }
+    }
+
+    // Deactivate players without a client
+    for (Player* player : mPlayers) {
```

Comment:
setting a player to inactive for even a second might have surprising results no? I wonder if this should be looking up in client_ids to decide whether to set player to inactive or active. It's inefficient at the moment because it's a vector, so no easy way to check membership efficiently. But the # of players is small so it doesn't really matter if we're doing O(n) lookups. eventually might want to consider an ordered hash.

Reply:
I was thinking of the case where a different thread might read this state

Reply:
That can't happen because we have the `mAcceptingNetworkGameObjectsReads` variable in `NetworkingServer::setState`

Reply:
yeah... it's an implicit dependency. code changes it's weird that a variable called `mAcceptingNetworkGameObjectReads` is gating something about players. in fact the NGO reads are coming over the UDP socket, the player data is more about the TCP registration/connection (that's where players are removed when the TCP connection dies)